File: java/src/main/java/com/alxarafe/app/domain/inventory/valueobject/BatchCode.java
package com.alxarafe.app.domain.inventory.valueobject;

import java.util.Objects;

public record BatchCode(String value) {

    public BatchCode {
        Objects.requireNonNull(value, "BatchCode cannot be null.");
        if (value.isBlank()) {
            throw new IllegalArgumentException("BatchCode cannot be empty.");
        }
        if (value.length() > 30) {
            throw new IllegalArgumentException(
                    "BatchCode must not exceed 30 characters. Got: " + value);
        }
    }

    @Override
    public String toString() {
        return value;
    }
}

File: php/src/Domain/Inventory/ValueObject/BatchCode.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Domain\Inventory\ValueObject;

use InvalidArgumentException;

final readonly class BatchCode
{
    public function __construct(private string $value)
    {
        if ($value === '') {
            throw new InvalidArgumentException('BatchCode cannot be empty.');
        }
        if (strlen($value) > 30) {
            throw new InvalidArgumentException(
                'BatchCode must not exceed 30 characters. Got: ' . $value
            );
        }
    }

    public function value(): string
    {
        return $this->value;
    }

    public function equals(self $other): bool
    {
        return $this->value === $other->value;
    }

    public function __toString(): string
    {
        return $this->value;
    }
}
